__tests__/app.test.js

```diff
@@ -12,3 +52,81 @@ describe("app behaviour", () => {
       })
   })
 })
+
+// // api test endpoint
+describe("/api", () => {
+  test(`GET responds 200 with an object`, () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        const { endpoints } = body
+        expect(typeof endpoints).toBe("object")
+      })
+  })
+  test(`GET responds 404 for an unknown endpoint`, () => {
+    return request(app)
+      .get("/api/nonexistent")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.endpoints).toBeUndefined()
+      })
+  })
+  test("API Endpoint Integration Test: verify GET request returns 200 status and response object Matches endpoints.json", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        const { endpoints } = body
+        expect(endpoints).toEqual(endpointsData)
+      })
+  })
+})
+
+// articles
+describe("/api/articles/", () => {
+  test("GET: 200 article object with correct properties", () => {
+    return request(app)
+      .get("/api/articles/10")
+      .expect(200)
+      .then(({ body }) => {
+        const { article } = body
+        expect(article).toMatchObject({
+          title: expect.any(String),
+          topic: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+        })
+      })
+  })
+  test("GET article 10", () => {
+    return request(app)
+      .get("/api/articles/10")
+      .expect(200)
+      .then(({ body }) => {
+        const { article } = body
+        const articleTen = {
+          article_id: 10,
+          title: "Seven inspirational thought leaders from Manchester UK",
+          topic: "mitch",
+          author: "rogersop",
+          body: "Who are we kidding, there is only one, and it's Mitch!",
+          created_at: "2020-05-14T04:15:00.000Z",
+          votes: 0,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        }
+        expect(article).toEqual(articleTen)
+      })
+  })
+  test("GET article with invalid ID should return 400", () => {
+    return request(app).get("/api/articles/notanumber").expect(400)
```

Comment:
these should both have a msg on the body of the response, "not found" and "bad request" or something similar